src/main/java/net/countercraft/movecraft/combat/features/directors/ArrowDirectors.java

```diff
@@ -95,13 +96,15 @@ private void processArrow(@NotNull Arrow arrow) {
         double speed = arrowVector.length(); // store the speed to add it back in later, since all the values we will be using are "normalized", IE: have a speed of 1
         arrowVector = arrowVector.normalize(); // you normalize it for comparison with the new direction to see if we are trying to steer too far
 
-        Block targetBlock = DirectorUtils.getDirectorBlock(p, ArrowDirectorRange);
-        Vector targetVector;
-        if (targetBlock == null || targetBlock.getType().equals(Material.AIR)) // the player is looking at nothing, shoot in that general direction
-            targetVector = p.getLocation().getDirection();
-        else { // shoot directly at the block the player is looking at (IE: with convergence)
-            targetVector = targetBlock.getLocation().toVector().subtract(arrow.getLocation().toVector());
-            targetVector = targetVector.normalize();
+        // the player is looking at nothing, shoot in that general direction
+        Vector targetVector = p.getLocation().getDirection();
+        if (ArrowDirectorRange >= 0) {
+            Block targetBlock = DirectorUtils.getDirectorBlock(p, ArrowDirectorRange);
+            if (targetBlock != null && !targetBlock.getType().equals(Material.AIR)) {
+                // shoot directly at the block the player is looking at (IE: with convergence)
+                targetVector = targetBlock.getLocation().toVector().subtract(arrow.getLocation().toVector());
+                targetVector = targetVector.normalize();
+            }
```

Comment:
And here

Reply:
Because DirectorUtils will ALWAYS execute a raycast, regardless of what you give it, sure the relevant code here could be included there however.